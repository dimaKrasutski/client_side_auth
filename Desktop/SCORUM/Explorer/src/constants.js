import * as THEME_CONSTANTS from 'theme/constants';

export const {
  LOCALE_KEYS,
  LANG_KEYS,
  LOCALE,
  LOCALES,
  DEFAULT_LOCALE,
  DEFAULT_LANG,
} = THEME_CONSTANTS;

export const UNDEFINED_ERROR_ID = 'explorer.error.unknown';

export const MIN_FRACTION_DIGITS = 3;
export const MAX_FRACTION_DIGITS = 9;

export const CURRENCY_USD = 'USD';
export const CURRENCY_EUR = 'EUR';
export const ALL_CURRENCY = [CURRENCY_USD, CURRENCY_EUR];
