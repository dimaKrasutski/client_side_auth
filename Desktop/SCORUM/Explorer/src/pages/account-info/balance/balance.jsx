import React from 'react';
import { pick } from 'ramda';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import classNames from 'classnames';
import Row from 'reactstrap/lib/Row';
import Col from 'reactstrap/lib/Col';
import { IconScr } from 'theme/components/icons/icon-scr/icon-scr';
import { IconSp } from 'theme/components/icons/icon-sp/icon-sp';
import { BalanceCard } from '../../../components/balance-card/balance-card';
import { BalanceCardBody } from '../../../components/balance-card/balance-card-body';
import './balance.scss';

export const BalanceUI = ({
  account, locale, className,
}) => {
  const currency = 'USD';
  const { balance, scorumpower } = pick(['balance', 'scorumpower'], account);

  return (
    <div className={classNames('balance-wrapper', className)}>
      <div className="balance">
        <Row className="no-gutters mb-1_5">
          <Col xs="6">
            <BalanceCard classNames="card-left">
              <BalanceCardBody
                title="explorer.account-info.balanse-card.title-scr"
                icon={<IconScr />}
                balance={balance}
                trades={1}
                currency={currency}
                locale={locale}
              />
            </BalanceCard>
          </Col>
          <Col xs="6">
            <BalanceCard classNames="card-right">
              <BalanceCardBody
                title="explorer.account-info.balanse-card.title-sp"
                icon={<IconSp />}
                balance={scorumpower}
                trades={1}
                currency={currency}
                locale={locale}
              />
            </BalanceCard>
          </Col>
        </Row>
      </div>
    </div>
  );
};

BalanceUI.propTypes = {
  locale: PropTypes.string.isRequired,
  account: PropTypes.objectOf(PropTypes.string),
  className: PropTypes.string,
};

BalanceUI.defaultProps = {
  account: {},
  className: null,
};

export const mapStateToProps = ({ accountInfo, intl }) => ({
  account: accountInfo.account,
  locale: intl.localeFull,
});

export const Balance = connect(mapStateToProps)(BalanceUI);
