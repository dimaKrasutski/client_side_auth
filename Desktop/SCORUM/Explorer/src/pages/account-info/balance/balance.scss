$balance-col-padding: ($spacer / 2) * .75;

.balance-wrapper {
  scroll-behavior: smooth;
  overflow-x: auto;

  .balance {
    min-width: 518px + ($grid-gutter-width-xs * 2);

    @media (max-width: 540px) {
      padding-right: $grid-gutter-width-xs;
      padding-left: $grid-gutter-width-xs;
    }

    .card {
      &-left {
        padding-right: $balance-col-padding;
      }

      &-right {
        padding-left: $balance-col-padding;
      }
    }

    .pub-key {
      word-wrap: break-word;
    }
  }
}
