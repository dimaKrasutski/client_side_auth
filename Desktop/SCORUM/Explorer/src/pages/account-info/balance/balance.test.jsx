import React from 'react';
import { BalanceUI, mapStateToProps } from './balance';
import { DEFAULT_LOCALE, CURRENCY_USD } from '../../../constants';

describe('<Balance />', () => {
  describe('<BalanceUI />', () => {
    test('should match snapshot', () => {
      const wrapper = shallow((
        <BalanceUI
          balance={100}
          scorumpower={100}
          trades={1}
          currency={CURRENCY_USD}
          locale={DEFAULT_LOCALE}
        />
      ));
      expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot no balance', () => {
      const wrapper = shallow(<BalanceUI locale={DEFAULT_LOCALE} currency={CURRENCY_USD} className="some-test" />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    const state = {
      accountInfo: {
        account: {
          balance: 100,
          scorumpower: 100,
        },
      },
      intl: {
        locale: 'en',
      },
    };

    test('should get accountInfo', () => {
      expect(mapStateToProps(state).account.balance).toEqual(state.accountInfo.account.balance);
    });

    test('should get balanceInfo', () => {
      expect(mapStateToProps(state).account.scorumpower).toEqual(state.accountInfo.account.scorumpower);
    });
  });
});
