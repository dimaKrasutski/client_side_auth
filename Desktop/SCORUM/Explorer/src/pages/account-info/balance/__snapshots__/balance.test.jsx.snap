// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Balance /> <BalanceUI /> should match snapshot 1`] = `
<div
  className="balance-wrapper"
>
  <div
    className="balance"
  >
    <Row
      className="no-gutters mb-1_5"
      tag="div"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="6"
      >
        <BalanceCard
          classNames="card-left"
          innerClassName={null}
        >
          <BalanceCardBody
            balance={null}
            currency="USD"
            icon={<IconScr />}
            locale="en-us"
            title="explorer.account-info.balanse-card.title-scr"
            trades={1}
          />
        </BalanceCard>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="6"
      >
        <BalanceCard
          classNames="card-right"
          innerClassName={null}
        >
          <BalanceCardBody
            balance={null}
            currency="USD"
            icon={<IconSp />}
            locale="en-us"
            title="explorer.account-info.balanse-card.title-sp"
            trades={1}
          />
        </BalanceCard>
      </Col>
    </Row>
  </div>
</div>
`;

exports[`<Balance /> <BalanceUI /> should match snapshot no balance 1`] = `
<div
  className="balance-wrapper some-test"
>
  <div
    className="balance"
  >
    <Row
      className="no-gutters mb-1_5"
      tag="div"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="6"
      >
        <BalanceCard
          classNames="card-left"
          innerClassName={null}
        >
          <BalanceCardBody
            balance={null}
            currency="USD"
            icon={<IconScr />}
            locale="en-us"
            title="explorer.account-info.balanse-card.title-scr"
            trades={1}
          />
        </BalanceCard>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="6"
      >
        <BalanceCard
          classNames="card-right"
          innerClassName={null}
        >
          <BalanceCardBody
            balance={null}
            currency="USD"
            icon={<IconSp />}
            locale="en-us"
            title="explorer.account-info.balanse-card.title-sp"
            trades={1}
          />
        </BalanceCard>
      </Col>
    </Row>
  </div>
</div>
`;
